### ISSUE TITLE
Fail whole test immediately when expectation fails

### ISSUE THREAD

**Issue by ndbroadbent** [Quick/Nimble#69]
Hi,

I just started using Quick + Nimble, and I noticed that all my expectations are run even when the first one fails. Is there a way make the test crash after the first failure?

**Comment by rbeeger** [comment]
I think creating a custom AssertionHandler won't help here. If I get this right, an AssertionHandler only decides how or if a failure is displayed or passed to XCTest, but it cannot prevent the evaluation of any other expectations after a failed one.

The behavior asked for in this request would immediately break out of the current it and resume execution on the next it, context or whatever comes next.

The only way for an AssertionHandler to do this would be to throw an exception, but then https://github.com/Quick/Quick/blob/master/Quick/Example.swift#L68 would need to somehow handle it. Otherwise it would break the whole test run.

**Comment by jeffh** [comment]
Ah I was thinking it was about rspec's `--fail-fast` option. My bad.

This just requires setting `continueAfterFailure = true` on the XCTestCase. Quick doesn't official support this, but it's not hard to do manually:

```swift
class MySpec: QuickSpec {
    override func setUp() {
         continueAfterFailure = false
    }
    override func spec() {
         it("fails only once") {
             fail("Only I appear")
             fail("And I don't")
         }
    }
}
```

I filed an issue on Quick for an official way to support feature: https://github.com/Quick/Quick/issues/249.

Closing this issue, continue the conversation on that issue.
